src/transport/heat_model.cc

```diff
@@ -300,6 +364,67 @@ void HeatTransferModel::compute_dirichlet_bc(const std::vector<arma::vec3> &poin
 		bc_values[i] = bc_value[i];
 }
 
+void HeatTransferModel::get_total_flux_bc_data(const std::vector<arma::vec3> &point_list,
+		const ElementAccessor<3> &ele_acc,
+		std::vector< arma::vec > &bc_flux,
+		std::vector< arma::vec > &bc_ad_value,
+		std::vector< arma::vec > &bc_sigma,
+		std::vector< arma::vec > &bc_ref_value)
+{
+	const unsigned int np = point_list.size();
+	vector<double> bc_flux_scalar(np),
+			bc_ad_temperature(np),
+			bc_robin_sigma(np),
+			bc_temperature(np);
+
+	data().bc_flux.value_list(point_list, ele_acc, bc_flux_scalar);
+	data().bc_ad_temperature.value_list(point_list, ele_acc, bc_ad_temperature);
+	data().bc_robin_sigma.value_list(point_list, ele_acc, bc_robin_sigma);
+	data().bc_temperature.value_list(point_list, ele_acc, bc_temperature);
+
+	for (unsigned int i=0; i<np; i++)
+	{
+		bc_flux[i] = bc_flux_scalar[i];
+		bc_ad_value[i] = bc_ad_temperature[i];
+		bc_sigma[i] = bc_robin_sigma[i];
+		bc_ref_value[i] = bc_temperature[i];
+	}
+}
+
+void HeatTransferModel::get_diffusive_flux_bc_data(const std::vector<arma::vec3> &point_list,
```

Comment:
Do not try to make fields of heat model scalar at this time. This makes lot of boiler plate and could be probably avoided when we use multifields (we have to do that soon, should discus it).